Replace current WSGIMiddleware implementation by a2wsgi one (#1825)
Kludex/uvicorn: docs/index.md

    })
```

## The WSGI Support

Uvicorn offer support to WSGI applications throught `WSGIMiddleware` which converts

> your WSGI into ASGI application seamlessly just by running it with `--interface wsgi`

Reviewer: Is this the first mention to the `WSGIMiddleware` on the documentation? I think this is just an implementation detail.

Author: Also cut it and put just the mention in the settings like you said

!!! warning

>     The present WSGIMiddleware will be deprecated in favour of [a2wsgi](https://github.com/abersheeran/a2wsgi).
>     If you are using Uvicorn to run WSGI applications, please add it as part of your project requirements.

Reviewer: It will be **removed**, but this PR already deprecates it.
```suggestion
    The present WSGIMiddleware is deprecated in favour of [a2wsgi](https://github.com/abersheeran/a2wsgi).
    If you are using Uvicorn to run WSGI applications, please add it as part of your project requirements.
```

Author: Yup, updated the docs

To use the WSGI interface install its dependency:

```shell
pip install a2wsgi
```

Create an application, in `example.py`:

```python
def app(environ, start_response):
    start_response('200 OK', [('Content-Type', 'text/plain')])
    return [b'Hello World!']
```

Run the server:

```shell

> $ uvicorn --interface wsgi example:app

Reviewer: Is the `--interface` needed, or does uvicorn infers it? I never used WSGI with uvicorn. :thinking:

Author: It fails to infer it, maybe a bug but it falls into asgi2 case that also uses a function instead of a coroutine in the config code:
```python
...
            elif inspect.isfunction(self.loaded_app):
                use_asgi_3 = asyncio.iscoroutinefunction(self.loaded_app)
...
```
